components/Dialogs/tests/snapshot/MDCAlertControllerLocalizationTests.m

```diff
@@ -0,0 +1,111 @@
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MaterialSnapshot.h"
+
+#import "MaterialDialogs.h"
+
+static NSString *const kTitleUrdu = @"عنوان";
+static NSString *const kMessageUrdu =
+    @"براہ کرم اپنا نیٹ ورک کنکشن چیک کریں اور دوبارہ کوشش کریں۔براہ کرم اپنا نیٹ ورک کنکشن چیک "
+    @"کریں اور دوبارہ کوشش کریں۔";
+
+static NSString *const kActionMediumUrdu = @"درمیانی";
+static NSString *const kActionLowUrdu = @"کم";
+
+@interface MDCAlertControllerLocalizationSnapshotTests : MDCSnapshotTestCase
+@property(nonatomic, strong) MDCAlertController *alertController;
+@property(nonatomic, strong) UIImage *iconImage;
+@end
+
+@implementation MDCAlertControllerLocalizationSnapshotTests
+
+- (void)setUp {
+  [super setUp];
+
+  // Uncomment below to recreate all the goldens (or add the following line to the specific
+  // test you wish to recreate the golden for).
+  // self.recordMode = YES;
+
+  self.iconImage = [[UIImage mdc_testImageOfSize:CGSizeMake(40, 40)]
+      imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate];
+  self.alertController = [[MDCAlertController alloc] init];
+  self.alertController.view.bounds = CGRectMake(0, 0, 300, 300);
+}
+
+- (void)tearDown {
+  self.alertController = nil;
+  self.iconImage = nil;
+
+  [super tearDown];
+}
+
+- (void)generateSnapshotAndVerifyForView:(UIView *)view {
+  [view layoutIfNeeded];
+
+  UIView *snapshotView = [view mdc_addToBackgroundView];
+  [self snapshotVerifyView:snapshotView];
+}
+
+- (void)changeViewToRTL:(UIView *)view {
+  if (@available(iOS 9.0, *)) {
+    view.semanticContentAttribute = UISemanticContentAttributeForceRightToLeft;
+    for (UIView *subview in view.subviews) {
+      [self changeViewToRTL:subview];
+    }
+  }
+}
+
+- (void)changeToRTL:(MDCAlertController *)alertController {
+  if (@available(iOS 9.0, *)) {
+    [self changeViewToRTL:alertController.view];
+  }
+}
+
+#pragma mark - Tests
+
+- (void)testPreferredContentSizeWithUrduLargeFont {
+  // When
+  self.alertController.title = kTitleUrdu;
+  self.alertController.titleIcon = self.iconImage;
+  self.alertController.message = kMessageUrdu;
+  NSString *urduFontName = @"NotoNastaliqUrdu";
+  UIFont *dialogBodyFont;
+  UIFont *dialogButtonFont;
+  if (@available(iOS 11, *)) {
+    // Noto Nastaliq Urdu was added in iOS 11, and is an extremely tall
+    // font for any given nominal point size.
+    dialogBodyFont = [UIFont fontWithName:urduFontName size:20.0];
+    dialogButtonFont = [UIFont fontWithName:urduFontName size:26.0];
+  }
+  self.alertController.messageFont = dialogBodyFont;
```

Comment:
Please add a fall-back font. If it's important to use a "large font" as indicated by the method title, please add one for the non-iOS 11 path.

Reply:
Addressed in 9ed4cbe40464c1fa6c7f23b34948efd3d7a61fbc.